.github/workflows/check-code-quality.yml

```diff
@@ -32,21 +32,14 @@ jobs:
       with:
         severity: style
 
+    # There's a daily check for _all_ links in check-broken-links.yml,
+    # so this only checks for new/changed links in the PR
     - name: Check for broken links
-      run: | 
-        touch valid_urls.txt redirected_urls.txt invalid_urls.txt
-        find . -type f \( -name "*.txt" -o -name "*.md" -o -name "*.py" -o -name "*.rst" \) \
-          \( \! -name 'CHANGES.md' \) \( \! -name 'requirements.txt' \) \( \! -name 'requirements-freeze.txt' \) | 
-          xargs grep -Eo "\b(https?)://[-A-Za-z0-9+&@#/%?=~_|$!:,.;]*[A-Za-z0-9+&@#/%=~_|$]" | 
-          sed 's/:/;/' | 
-          xargs -n 1 -P 8 ./.github/workflows/check-url.sh
-
-    - name: Summarize broken link check
       run: |
-        NUM_OK=$(wc -l valid_urls.txt | cut -d " " -f 1)
-        NUM_REDIRECT=$(wc -l redirected_urls.txt | cut -d " " -f 1)
-        NUM_BAD=$(wc -l invalid_urls.txt | cut -d " " -f 1)
-        cat invalid_urls.txt
-        echo -en "========== \033[0;32m$NUM_OK passed\033[0;0m, \033[0;33m$NUM_REDIRECT redirected\033[0;0m, \033[0;31m$NUM_BAD failed\033[0;0m ==========\n"
-        exit $NUM_BAD
-
+        set -eu -o pipefail
+        git diff origin/master... |
+          (grep '^+' || (($? == 1)) ) |
+          (grep -Eo '\b(https?)://[-A-Za-z0-9+&@#/%?=~_|$!:,.;]*[A-Za-z0-9+&@#/%=~_|$]' || (($? == 1)) ) |
+          sed 's/:/;/' |
+          xargs -rn 1 .github/workflows/check-url.sh
+        test ! -s invalid_urls.txt
```

Comment:
Comparing this to the flake8 check (which also uses `git diff`)

- `flake8`: Diffs filenames only, filters for `.py` files, runs flake8 on filenames.
- `check-url`: Shows full diffs, filters for additions (`+`), filters added lines for URLs, then passes URLs to `check-url.sh`.
  - `-r` is used because the diff may not always contain URLs, `-P 8` is absent for reasons mentioned in my other comment.
  - `test ! -s` passes (returns `0`) if `NOT` "file size greater than 0", meaning... if the file size _is_ greater than 0, then test will fail and return code `1`. (And, `set -e` causes this to result in an error.)

Makes sense to me!

Reply:
Exactly!